oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderJwkSupport.java

```diff
@@ -119,8 +121,18 @@ public Jwt decode(String token) throws JwtException {
 
 			jwt = new Jwt(token, issuedAt, expiresAt, headers, jwtClaimsSet.getClaims());
 
+		} catch ( RemoteKeySourceException jwkFailure ) {
+			if ( jwkFailure.getCause() instanceof ParseException ) {
+				throw new JwtException("An error occurred while attempting to decode the Jwt: Malformed Jwk set");
```

Comment:
This exception message reveals the underlying jwk set or the contents of whatever page it is pointed at, so this catches and overrides it with a less-revealing message.